epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -1144,6 +1200,14 @@ def get_process_function_definitions(self, write=True):
             replace_dict['cipd2tipd'] = '//memcpy( cIPD, tIPD, %i * sizeof( fptype ) ); // nparam=0'%len(params)
             replace_dict['cipddump'] = ''
             replace_dict['cipdhrdcod'] = '//__device__ const fptype* cIPD = nullptr; // unused as nparam=0'
+        if self.model_name[:2] == 'sm' :
```

Comment:
same comment here ... 
to add details, it is likely better to trigger this depending on the presence/absence of parameter that can not be handle like the SM...

Reply:
Hm ok yes. I see what you mean, a bit what  was saying above
- do some extra "BSM-like" checks all the time, also in SM processes, and see what happens
- if some extra non-SM Parameters are identified, then do some special handling...

I need to try this out in practice.

This is a big change. I would try this out only directly on the second more comprehensive PR #824... or, even merge #824 and then do this in a later step...? Any preferences?